channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/dto/JaicpDialerAPI.kt

```diff
@@ -138,6 +218,21 @@ private fun checkLocalTime(data: JaicpDialerAPI.RedialData) {
     }
 }
 
+private fun checkAllowedTime(data: JaicpDialerAPI.RedialData) {
+    val formatter = DateTimeFormatter.ofPattern("HH:mm")
+
+    data.allowedTime?.intervals?.filterNotNull()
+        ?.flatten()
+        ?.forEach { interval ->
+            val start = interval.localTimeFrom.let { LocalTime.parse(it, formatter) }
+            val end = interval.localTimeTo.let { LocalTime.parse(it, formatter) }
+
+            require(start.isBefore(end)) {
+                "localTimeFrom cannot be higher then localTimeTo of allowedTime"
```

Comment:
`checkAllowedTime` will be in stacktrace, so I don't think we really should reference it in the error message. Also, I'd suggest changing the "lower-higher" relationship to "before-after" as we say about time.
```suggestion
                "localTimeFrom must be before localTimeTo"
```

Reply:
It's better